### PR TITLE
Upgrade lints to 3.0.0

### PR DESCRIPTION
**Pull request by isoos**
Upgrading lints affects both pana (issues fixed) and the analysis of packages.

### CODE REVIEW

File: lib/src/tag/tagger.dart [hunk 1]
```diff
@@ -70,6 +70,7 @@
 /// * platform:windows
 ///
 /// A package has the same platform tags as the primary library.
+library tagger;
```
Comment (parlough) [inline lib/src/tag/tagger.dart:73]: Consider dropping the library names (here and elsewhere):

```suggestion
library;
```

File: lib/src/tag/_violations.dart [hunk 2]
```diff
@@ -81,6 +81,8 @@
 /// If both "platforms" and "flutter.platforms" are declared "platforms" takes
 /// precedence. (There might be rare cases where you want to declare a more
 /// narrow set of platforms than you have plugin support for).
+library _violations;
```
Comment (parlough) [inline lib/src/tag/_violations.dart:84]:
```suggestion
library;
```

**Review by parlough — COMMENTED** [review]
Exciting!

**Review by szakarias — APPROVED** [review]
LGTM

**Pull request merged**